tests/integration/test_charm.py

```diff
@@ -361,3 +365,107 @@ def generate_s3_config(s3_url: str) -> Dict:
     s3_config["ip_address"] = s3_ip_address
     s3_config["endpoint"] = s3_endpoint
     return s3_config
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+@pytest.mark.requires_secrets
+@pytest.mark.usefixtures("setup_saml_config")
+async def test_saml_login(  # pylint: disable=too-many-locals
+    ops_test: OpsTest,
+    app: Application,
+    pytestconfig: pytest.Config,
+    requests_timeout: int,
+    run_action,
+):
+    """
+    arrange: after discourse charm has been deployed, with all required relation established.
+    act: add an admin user and enable force-https mode.
+    assert: user can login discourse using SAML Authentication.
+    """
+    assert ops_test.model
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+    email = pytestconfig.getoption("--saml-email")
```

Comment:
Should this be a fixture?

Reply:
This option is only used in this test case, put this in the fixture might complicated the error handling. In this test case, an error need to raise if the option is not provided, but for other test cases, missing such option is not an error. If keep the option extraction in the test case, I can raise an error message with the test case name in it. But if I put the option extraction in a fixture, the error message mentioning the required test cases name seems inappropriate.

Reply:
I'm guessing that pytest only invokes fixture that are actually required for a test run. So, if a run doesn't need the fixture with this option, then it wouldn't get executed and no error is raised. The error message in the fixture could be "the invoked tests require the <option> to be able to run"

Reply:
Okay, I update the code to put SAML arguments into a fixture.